### ISSUE TITLE
Using track() with Fieldset/Control and an Immutable store

### ISSUE THREAD

**Issue by marcandrews** [davidkpiano/react-redux-form#846]
### The Problem
`track()` does not work when used with an Immutable store to determine the model for a `Fieldset` component. It also does not work with `Control` components contained with the `Fieldset` component. Other RRF components may be affected.

### Steps to Reproduce
1. Use `react-redux-form/immutable` with an Immutable store.
2. Use track to determine the model for a `Fieldset` or `Control` component.

### Expected Behavior
`track()` should be able to determine the model for a `Fieldset` component.

### Actual Behavior
The following error occurs:
```
TypeError: Cannot read property 'id' of null
```

### Reproducible Code Example
See CodeSandbox [here][CodeSandbox]

[CodeSandbox]: https://codesandbox.io/s/76kpQXQz1

**Comment by marcandrews** [comment]
@davidkpiano thanks!

**Comment by marcandrews** [comment]
@davidkpiano I updated my CodeSandbox with 1.13.0. The error is gone, but the `Control.text` components assigned to retrieve the keys in the currently selected collection do not display the values, not can they update them. Any ideas?

**Comment by marcandrews** [comment]
@davidkpiano,

I've forked your repo and potentially found the problem: [`findKey()`](https://github.com/davidkpiano/react-redux-form/blob/master/src/utils/find-key.js#L5) is equipped to handle Immutable Lists. The following modifications appear to solve the issue; however, I did not test these changes thoroughly.

**/src/utils/find-key.js:**
```js
import { List } from 'immutable';

export default function findKey(object, predicate) {
  let resultKey;

  if (List.isList(object)) {
    resultKey = object.findKey(predicate);
  } else {
    Object.keys(object).some((key) => {
      const isKey = predicate(object[key], key, object);

      if (isKey) {
        resultKey = key;
        return true;
      }

      return false;
    });
  }

  return resultKey;
}

```

Let me know what you think or if you'd like me to submit a PR.

**Comment by davidkpiano** [comment]
Fixed by #938
